File: src/com/richard/walker/outlets/java/BritishOutlet.java
package com.richard.walker.outlets.java;

public class BritishOutlet implements EuropeanOutlet{

    private static final double WORKING_VOLTAGE = 220.00;

    @Override
    public Double provide220Volts(){
        System.out.println("British Outlet providing " + WORKING_VOLTAGE + "V of power!");
        return WORKING_VOLTAGE;
    }

    @Override
    public Boolean isFunctioning(){
        System.out.println("British Outlet is working.");
        if(WORKING_VOLTAGE >= (WORKING_VOLTAGE * 0.90)){
            return Boolean.TRUE;
        } else{
            return Boolean.FALSE;
        }
    }
}

File: src/com/richard/walker/outlets/kotlin/BritishOutlet.kt
package com.richard.walker.outlets.kotlin

private const val WORKING_VOLTAGE = 220.0
private const val isVoltageAcceptable = WORKING_VOLTAGE >= WORKING_VOLTAGE * .90

class BritishOutlet : EuropeanOutlet {
    override fun provide220Volts(): Double {
        println("British outlet is providing $WORKING_VOLTAGE V of power!")
        return WORKING_VOLTAGE
    }

    override fun isFunctioning(): Boolean = when (isVoltageAcceptable) {
        true -> {
            println("British outlet is working")
            isVoltageAcceptable
        }
        false -> {
            println("British outlet is not working")
            isVoltageAcceptable
        }
    }
}